src/components/Search/SearchDatePresetFilterBase.tsx

```diff
@@ -1,48 +1,60 @@
 import useLocalize from '@hooks/useLocalize';
-import useOnyx from '@hooks/useOnyx';
-import useThemeStyles from '@hooks/useThemeStyles';
-import {updateAdvancedFilters} from '@libs/actions/Search';
-import Navigation from '@libs/Navigation/Navigation';
 import {isSearchDatePreset} from '@libs/SearchQueryUtils';
-import type {SearchDateModifier, SearchDateModifierLower} from '@libs/SearchUIUtils';
+import type {SearchDateModifier} from '@libs/SearchUIUtils';
 import CONST from '@src/CONST';
-import type {TranslationPaths} from '@src/languages/types';
-import ONYXKEYS from '@src/ONYXKEYS';
-import ROUTES from '@src/ROUTES';
-import type {SearchDateFilterKeys, SearchDatePreset} from './types';
+import type {SearchDatePreset} from './types';
+
+type SearchDateValues = Record<SearchDateModifier, string | undefined>;
+
+type SearchDatePresetFilterBaseHandle = {
+    /** Gets date values */
+    getDateValues: () => SearchDateValues;
+
+    /** Clears date values */
+    clearDateValues: () => void;
+
+    /** Sets the date value of the selected date modifier to the ephemeral date value (the selected date in calendar) */
+    setDateValueOfSelectedDateModifier: () => void;
+
+    /** Clears the date value of the selected date modifier */
+    clearDateValueOfSelectedDateModifier: () => void;
+};
 
 type SearchDatePresetFilterBaseProps = {
-    /** Key used for the date filter */
-    dateKey: SearchDateFilterKeys;
+    /** Default date values */
+    defaultDateValues: SearchDateValues;
+
+    /** Selected date modifier */
+    selectedDateModifier: SearchDateModifier | null;
 
-    /** The translation key for the page title */
-    titleKey: TranslationPaths;
+    /** Callback when a date modifier is selected */
+    onSelectDateModifier: (dateModifier: SearchDateModifier | null) => void;
 
     /** The date presets */
     presets?: SearchDatePreset[];
+
+    /** The ref handle */
+    ref: Ref<SearchDatePresetFilterBaseHandle>;
 };
 
-function SearchDatePresetFilterBase({dateKey, titleKey, presets}: SearchDatePresetFilterBaseProps) {
-    const styles = useThemeStyles();
+/**
+ * SearchDatePresetFilterBase is a partially controlled component:
+ * - The selected date modifier is controlled.
+ * - The date values are uncontrolled. This is done to avoid duplicating the `setDateValue` logic and also to avoid exposting the `ephemeralDateValue` state.
+ *
+ * There are cases where the parent is required to alter the internal date values e.g. reset the values, in such cases you should use the ref handle.
+ * Typically you are expected to use this component with a save and a reset button.
+ * - On save: if a date modifier is selected (i.e. user clicked save at the calendar picker) you should `setDateValueOfSelectedDateModifier` otherwise `getDateValues`
+ * - On reset: if a date modifier is selected (i.e. user clicked reset at the calendar picker) you should `clearDateValueOfSelectedDateModifier` otherwise `clearDateValues`
+ */
```

Comment:
This may seem complicated but it's what allows us to reuse same logic in `DateSelectPopup` instead of copy-pasting (as it's the case in `main`)